### PR TITLE
Sscs 3256 removal of 2 pages pattern

### PR DESCRIPTION
**Pull request by barbaracassani**
- Add JS which allows the reasons for appealing fields to be on one page where the can 'Add another' on a single page rather than having to loop around on two pages.

- Add `add-reason` class which contains the functionality to post the values when submitted and get the values (if any) when coming back after submitting.
- Add nunjucks loaders to use nunjucks templates from look-and-feel in the JS class
- Add multiple functional tests and e2e tests
- ignore the non-JS functional tests for this page

### CODE REVIEW

File: assets/js/add-reason.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  addAnother() {
+    $('.add-another-add-link').click(event => {
+      event.preventDefault();
+      this.counter += 1;
+      this.addFields();
+    });
+  }
+
+  static startAddReason() {
+    return (window.location.pathname === '/reason-for-appealing');
```
Comment (barbaracassani) [inline assets/js/add-reason.js]: I know I wrote this line 😊 but anyway, maybe it's better to put the url in a constant.

Comment (bendiggle) [reply assets/js/add-reason.js]: Actually yeah, i should import it from paths.js

File: assets/js/add-reason.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  buildReasonForAppealingField(errors, value) {
+    return this.textareaField({
+      id: `${this.textareaId}-${this.counter}`,
+      errors: errors || [],
+      value: value || ''
+    }, 'Why you disagree with it', null, false, 'You can write as much as you want');
+  }
+
+  onSubmit() {
+    const self = this;
```
Comment (barbaracassani) [inline assets/js/add-reason.js]: I've just seen this. We ought to use `bind` instead really...

**Comment by bendiggle** [comment]
Found an issue with one per page. Occurs when deleting an item using Ajax (occurs on the dates cant attend page)

**Pull request merged**